gldemo/plugin.cpp

```diff
@@ -484,45 +402,10 @@ class gldemo : public threadloop {
 		vertexNormalAttr = glGetAttribLocation(demoShaderProgram, "vertexNormal");
 		modelViewAttr = glGetUniformLocation(demoShaderProgram, "u_modelview");
 		projectionAttr = glGetUniformLocation(demoShaderProgram, "u_projection");
-
 		colorUniform = glGetUniformLocation(demoShaderProgram, "u_color");
 
-		
-		groundObject.init(vertexPosAttr,
-			vertexNormalAttr,
-			colorUniform,
-			Ground_plane_NUM_TRIANGLES,
-			&(Ground_Plane_vertex_data[0]),
-			&(Ground_Plane_normal_data[0]),
-			GL_STATIC_DRAW
-		);
-
-		waterObject.init(vertexPosAttr,
-			vertexNormalAttr,
-			colorUniform,
-			Water_plane001_NUM_TRIANGLES,
-			&(Water_Plane001_vertex_data[0]),
-			&(Water_Plane001_normal_data[0]),
-			GL_STATIC_DRAW
-		);
-
-		treesObject.init(vertexPosAttr,
-			vertexNormalAttr,
-			colorUniform,
-			Trees_cone_NUM_TRIANGLES,
-			&(Trees_Cone_vertex_data[0]),
-			&(Trees_Cone_normal_data[0]),
-			GL_STATIC_DRAW
-		);
-
-		rocksObject.init(vertexPosAttr,
-			vertexNormalAttr,
-			colorUniform,
-			Rocks_plane002_NUM_TRIANGLES,
-			&(Rocks_Plane002_vertex_data[0]),
-			&(Rocks_Plane002_normal_data[0]),
-			GL_STATIC_DRAW
-		);
+		// Load/initialize the demo scene.
+		demoscene = new ObjScene("demo2.obj");
```

Comment:
Same here with the pointer.

Reply:
This doesn't work, because `ObjScene` does not have a default constructor, and C++ requires all members to have a default constructor... :/

`ObjScene` is constructed with the filename associated with the .obj file. This is obtained through reading envvars, which... can't be safely done with error checking in an initializer list.

Reply:
OK, I made a default constructor. All done.